moon/controller_excavator_round2.py

```diff
@@ -347,47 +387,29 @@ def vslam_reset_time(response):
                         print(self.sim_time, self.robot_name, "Vehicle attitude: %.2f" % attitude)
                     self.publish("desired_movement", [GO_STRAIGHT, math.degrees(attitude * 100), 0.1])
                     self.wait(timedelta(milliseconds=100))
-                except BusShutdownException:
-                    raise
                 except MoonException as e:
                     print(self.sim_time, self.robot_name, "Exception while waiting for excavator to settle: ", str(e))
             self.publish("desired_movement", [0, 0, 0])
 
-            while self.mount_angle is None or abs(normalizeAnglePIPI(math.pi - self.mount_angle)) > 0.2:
-                try:
-                    self.wait(timedelta(seconds=1))
-                except BusShutdownException:
-                    raise
-                except MoonException as e:
-                    print(self.sim_time, self.robot_name, "Exception while waiting for arm to align: ", str(e))
-
+            self.wait_for_arm(math.pi)
```

Comment:
is it expected to have this unprotected? (try/except)

Reply:
I have added the protection within the function itself (line ~347).  I was getting tired of having these 10-line blocks everywhere. Please check inside the function to make sure it makes sense.